amigo.go

```diff
@@ -22,8 +22,22 @@ var (
 type handlerFunc func(map[string]string)
 type eventHandlerFunc func(string)
 
+type Amigo interface {
+	CapitalizeProps(c bool)
+	Action(action map[string]string) (map[string]string, error)
+	AgiAction(channel, command string) (map[string]string, error)
+	Connect()
+	Connected() bool
+	On(event string, handler func(string))
+	RegisterDefaultHandler(f handlerFunc) error
+	RegisterHandler(event string, f handlerFunc) error
+	SetEventChannel(c chan map[string]string)
+	UnregisterDefaultHandler(f handlerFunc) error
+	UnregisterHandler(event string, f handlerFunc) error
+}
+
 // Amigo is a main package struct
-type Amigo struct {
+type AmigoImpl struct {
```

Comment:
So this is an implementation of **Amigo** interface. May be better to name it **Asterisk** or **AmigoAsterisk** instead of **AmigoImpl**?